pkg/telemetry/tracing.go

```diff
@@ -19,9 +19,10 @@ import (
 
 type TracerOption func(d *customTracer)
 
-func WithOTLPEndpoint(endpoint string) TracerOption {
+func WithOTLPEndpoint(endpoint string, insecure bool) TracerOption {
```

Comment:
Please don't overload this Trace option. Make a unique one for this individual option.

e.g.

```go
func WithOTLPInsecure() TracerOption {
    return func(d *customTracer) {
        d.insecure = true
    }
}
```

Reply:
Done, thanks!